dask_searchcv/adaptive.py

```diff
@@ -0,0 +1,538 @@
+    data : dict
+        Data to train/validate on. {'train': [X, y], 'val': [X_val, y_val]}
+    eta : int
+        This controls how aggressive the "killing" of models should be. The
+        "infinite time" theory suggest eta == np.e, about 3.
+    n : int
+        Number of models to train initially
+    r : int
+        Maximum number of times to call partial_fit on any one model
+    s : int
+        The bracket. Used for intermediate variables
+    _prefix : str
+        A prefix to make the IDs for this call to _successive_halving unique
+        from other brackets
+    dry_run : bool
+        Whether this is a dry run or not. If it is, do not train the model and
+        allow the score to be random. Setting ``dry_run=True`` is useful for
+        collecting stats.
+    scorer : callable
+        The function used to score any model.
+
+    Returns
+    -------
+    results : dict, {str: value}
+        Results of this call to _successive_halving. Has values of
+
+        history : list
+            The history of every call to the model. This will be a list of dicts,
+            and eventually found in self.history. Has keys of bracket,
+            bracket_iter, val_score, model_id, partial_fit_iters, num_models as
+            well as all the parameter keys.
+        final_val_scores :
+
+
+    """
+    if any(x is None for x in [n, r, s, model, scorer]):
+        raise ValueError('n, r, and s are required')
+
+    best = {'val_score': -np.inf, 'config': {}}
+
+    ids = [_prefix + '-' + str(i) for i in range(n)]
+    params = ParameterSampler(params, len(ids))
+    params = {k: param for k, param in zip(ids, params)}
+    models = {}
+    final_val_scores = {k: -np.inf for k in ids}
+    for k, param in params.items():
+        models[k] = clone(model).set_params(**param)
+
+    history = []
+    iters = 0
+    times = []
+    for i in range(s + 1):
+        n_i = math.floor(n * eta**-i)
+        r_i = r * eta**i
+        iters += r_i
+        msg = ('Training %d models for %d iterations during iteration %d '
+               'of bracket=%d')
+        logger.info(msg, n_i, r_i, i, s)
+
+        if debug:
+            results = {k: _train(model, data, max_iter=r_i, s=s, i=i, k=k,
+                                 dry_run=dry_run, scorer=scorer, **fit_kwargs)
+                       for k, model in models.items()}
+        else:
+            delayed_results = {k: dask.delayed(_train)(model, data, max_iter=r_i,
+                                                       s=s, i=i, k=k, dry_run=dry_run,
+                                                       scorer=scorer, **fit_kwargs)
+                               for k, model in models.items()}
+            results = {k: v.compute() for k, v in delayed_results.items()}
+            #  results = dask.compute(delayed_results)[0]
+
+        val_scores = {k: r[0] for k, r in results.items()}
+        times += [{'id': k, **r[1]} for k, r in results.items()]
```

Comment:
Looks like this is invalid syntax.

Reply:
Yup. I picked it up at the same time, and it's fixed in the most recent commit.

There's a bigger issue for the failing tests – dask-searchcv doesn't depend on dask-ml. That should be resolved in the most recent commit, which implements a `cross_validation_score`.